src/components/views/elements/SettingsFlag.tsx

```diff
@@ -21,58 +21,76 @@ import createReactClass from 'create-react-class';
 import SettingsStore from "../../../settings/SettingsStore";
 import { _t } from '../../../languageHandler';
 import ToggleSwitch from "./ToggleSwitch";
+import StyledCheckbox from "./StyledCheckbox";
 
-export default createReactClass({
-    displayName: 'SettingsFlag',
-    propTypes: {
-        name: PropTypes.string.isRequired,
-        level: PropTypes.string.isRequired,
-        roomId: PropTypes.string, // for per-room settings
-        label: PropTypes.string, // untranslated
-        onChange: PropTypes.func,
-        isExplicit: PropTypes.bool,
-    },
+interface IProps {
+    name: string,
+    level: string,
+    roomId?: string, // for per-room settings
+    label?: string, // untranslated
+    isExplicit: boolean,
+    // XXX: once design replaces all toggles make this the default
+    useCheckbox?: boolean,
+    onChange(checked: boolean): void,
+}
 
-    getInitialState: function() {
-        return {
+interface IState {
+    // XXX: make this generic when the settings store is typed
```

Comment:
You could still make it Generic now and just use `<T = any>` so that anyone not specifying a type gets the any behaviour.